### PR TITLE
POC: keep live timeseries moving left

### PR DESCRIPTION
**Pull request by ryantxu**
When looking at live data, we want the graphs to keep moving left regardless of the last query response.  The current behavior updates with each packet -- this adds an additional timer to keep updating the timeRange 

Example with 2s data, but refreshing at 10hz:
20sec

- [ ] panel options? dashboard/context option?
- [ ] other x=time panels? or in panel chrome?
- [ ] selective HOC wrapper?  currently always wrapped, can it be selective?
- [ ] calculate a reasonable update rate based on pixels

### CODE REVIEW

**Comment by ryantxu** [comment]
cc @dejapong

File: packages/grafana-ui/src/components/TimeSeries/TimeSeries.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { preparePlotConfigBuilder } from './utils';
 import { withTheme2 } from '../../themes/ThemeContext';
 import { PanelContext, PanelContextRoot } from '../PanelChrome/PanelContext';
 
-const propsToDiff: string[] = [];
+const propsToDiff: string[] = ['timeRange'];
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/TimeSeries/TimeSeries.tsx:11]: hymmm -- I guess this invalidates the whole graph.  maybe we need `GraphNG` to check timeRange explicitly

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 2]
```diff
@@ -135,6 +138,28 @@ export class PanelChrome extends Component<Props, State> {
           next: (data) => this.onDataUpdate(data),
         })
     );
+
+    // TODO? check if the panel has X=time?
+    // Note these events will only be broadcast if the dashboard is enabled
+    if (this.props.plugin) {
+      this.subs.add(
+        this.props.dashboard.events.getStream(LiveDashboardTick).subscribe({
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx:146]: Use subscribe method on the event bus instead, getStream is only if you need to use Observable operators (map / merge etc)

**Review by torkelo — COMMENTED** [review]
The LiveRefreshComponent looks nice in that it abstracts the logic away from the TimeSeries component, but wonder if that logic rather belongs in PanelChrome? or somehow so we can share it between panels. 

The thing that feels off here is the logic added to DashhboardPage, it feels like it could be moved to TimeSrv or some other service/component (DashboardPage does too much already).

**Comment by leeoniya** [comment]
rather than setting up an additional interval/timer, i'd probably prefer to hook into our existing 10hz live/perf interval [1] via something like `liveInterval.register(chart1RangeUpdater)`. having multiple rapid timers ticking out of sync is not ideal.

[1] https://github.com/grafana/grafana/blob/main/public/app/features/live/perf.ts

**Comment by ryantxu** [comment]
Closing in favor of https://github.com/grafana/grafana/pull/37769

**Pull request closed without merging**